sound/soc/sof/sof-audio.c

```diff
@@ -468,24 +468,24 @@ int sof_machine_check(struct snd_sof_dev *sdev)
 	struct snd_sof_pdata *sof_pdata = sdev->pdata;
 	const struct sof_dev_desc *desc = sof_pdata->desc;
 	struct snd_soc_acpi_mach *mach;
-	int ret;
 
-#if !IS_ENABLED(CONFIG_SND_SOC_SOF_FORCE_NOCODEC_MODE)
+	if (!IS_ENABLED(CONFIG_SND_SOC_SOF_FORCE_NOCODEC_MODE)) {
 
-	/* find machine */
-	snd_sof_machine_select(sdev);
-	if (sof_pdata->machine) {
-		snd_sof_set_mach_params(sof_pdata->machine, sdev);
-		return 0;
+		/* find machine */
```

Comment:
This is really nice. Would we actually now be able to support a module param to affect this check? It's a very common usage that if codecs or mach driver don't work for some reason, you want to check the DSP part first with nocodec, and it's a drag that you have to rebuild the kernel for this. Being able to do this with a module param would be even better.

Reply:
> you want to check the DSP part first with nocodec, and it's a drag that you have to rebuild the kernel for this. 

+1000

Reply:
several reasons:

a) there's a known bug that if we enable HDMI but don't use it the pm_runtime gets confused.
b) we want to preserve code that does not use all the snd_hdac stuff and might be reusable in non GPL contexts.
c) we would always find HDMI so would always use the idisp topology with HDMI only.